internal/keepers/filter.go

```diff
@@ -121,19 +113,29 @@ func (rc *reportCoordinator) Accept(key types.UpkeepKey) error {
 		return err
 	}
 
+	// Set the key as accepted within activeKeys
 	rc.activeKeys.Set(key.String(), false, util.DefaultCacheExpiration)
 
+	// Also apply an idBlock if the key is after an existing idBlock check key
 	blockKey, _, err := key.BlockKeyAndUpkeepID()
 	if err != nil {
 		return err
 	}
 
-	if bl, ok := rc.idBlocks.Get(string(id)); ok && string(bl.KeyBlockNumber) > string(blockKey) {
-		return nil
+	bl, ok := rc.idBlocks.Get(string(id))
```

Comment:
Doing `string(id)` here still invites ids like `00001` into the comparison because the input is not getting validated anywhere. We have explicit json parsing implementations for keys, but the observation was changed to an array of identifiers. 

This needs to follow the same paradigm as `UpkeepKey` where the type or interface implements `fmt.Stringer` AND it needs explicit parsing and validation functions.

Reply:
The core issue we need to check is validation of observations which we get from nodes. I'll fix that in this PR. 
Our overall types (concrete and interfaces) and getting really messed up and I am not sure that's the correct way to go about converting all types to interfaces. Will do a thorough cleanup of that in a separate PR

Reply:
The observation validation looks good. Thanks!